### DISCUSSION TITLE
GET /payments/payment-operations Please return a unique identifier such as id in this api

### DISCUSSION THREAD

**Discussion by a785817258** [allegro/allegro-api#9567] [Suggestions - rejected]
GET /payments/payment-operations Please return a unique identifier such as id in this api

**Comment by Lukasz-Zurek** [comment]
Regarding IDs in [GET /payments/payment-operations](https://developer.allegro.pl/documentation/#tag/Payments/operation/getPaymentsOperationHistory):

Please note that **there is no single, universal ID field for all transaction types**. The specific ID field returned depends directly on the transaction type:

- PAYMENT / PAYMENT_REFUND: Uses **payment.id** (UUID of the buyer's payment or original transaction).
- DEDUCTION / CHARGE: Uses **deduction.id** (identifies the specific fee deduction from your balance).
- PAYOUT / PAYOUT_CANCEL: Uses **payout.id** (identifies the bank payout).
- Other operations (CORRECTION, BLOCKADE, etc.): Uses the top-level id field.

**Best Practice**

Always check the type field first instead of relying on a single ID, as payment, deduction, and payout objects are mutually exclusive and present only when applicable to that operation type.

For full details, check the [Allegro REST API documentation](https://developer.allegro.pl/documentation/?utm_source=gemini#tag/Payments/operation/getPaymentsOperationHistory).

With that in mind, I'll go ahead and close this request. Thanks for reaching out!
